modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -48,5 +58,53 @@ object ConstructionService {
 
       NonEmptyList.fromList(accountIdentifiers)
     }
+
+    def transactionToOperations(transaction: Transaction): Either[ConstructionError, NonEmptyList[Operation]] = {
+      type Refinement = Not[Equal[0L]]
+
+      val negativeTransfer = refineV[Refinement](-transaction.amount.value.value).map(
+        (transaction.source, _, OperationIndex(0L))
+      )
+      val positiveTransfer = refineV[Refinement](+transaction.amount.value.value).map(
+        (transaction.destination, _, OperationIndex(1L))
+      )
+
+      NonEmptyList.of(negativeTransfer, positiveTransfer).traverse {
+        case (Right((address, amount, operationIndex))) =>
+          Operation(
+            OperationIdentifier(operationIndex),
+            none,
+            OperationType.Transfer,
+            none,
+            AccountIdentifier(address, none).some,
+            Amount(AmountValue(amount), DAG).some
+          ).asRight[ConstructionError]
+        case _ => Left(MalformedTransaction)
+      }
+    }
+
+    def parseSignedTransaction(hex: Hex): EitherT[F, ConstructionError, ConstructionParse.ParseResult] = {
+      val response = for {
+        signedTransaction <- KryoSerializer[F].deserialize[Signed[Transaction]](hex.toBytes).toEitherT
+        operations <- transactionToOperations(signedTransaction).toEitherT
+        proofs <- signedTransaction.proofs.toNonEmptyList.traverse(_.id.hex.toPublicKey).attemptT
+        accountIds = proofs.map(_.toAddress).map(AccountIdentifier(_, none))
+      } yield ConstructionParse.ParseResult(operations, accountIds.some)
+
+      response.leftMap(_ => MalformedTransaction)
```

Comment:
why `response`? It's not a response in here

Reply:
Renamed to `result`.